### PR TITLE
TMEDIA 716 refactor ads block v2

### PR DESCRIPTION
**Pull request by JackHowa**
## Description

Refactor ads block 

## Jira Ticket

- [TMEDIA-716](https://arcpublishing.atlassian.net/browse/TMEDIA-716)

## Acceptance Criteria

Google Ad (ads-block) block is migrated to the new architecture.

### Storybook 

1. Look at chromatic
2. Open the ads block. 
3. See the admin view story
4. See the label hidden and displayed in the other ad stories

### Fusion 

1. Checkout this branch `git checkout TMEDIA-716-ads-block-v2-refactor`
5. Use the feature pack branch as well `TMEDIA-716-ads-block-v2-refactor`
6. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/ads-block`
7. Create a new page with test-ad 
8. Add an ads block to the page 
9. Set the label to "cube"
10. See the placeholder container in the admin with the dimensions and name 
11. Open the eye icon on the page. If the page is not immediately there, refresh until it is. 
12. See the label "ADVERTISEMENT" above the ad
13. Go back to the admin
14. Set to hide the label. 
15. Go to the published page
16. See "ADVERTISEMENT" is no longer present

### Before 

- admin storybook 
https://5f91e4721ccaba0022a10782-qucvrsbzrw.chromatic.com/?path=/story/blocks-ads-block--admin-view
Screen Shot 2022-08-02 at 17 12 43

#### reader view 
- using `before` with styled components to inline the text with the dynamic width ad 
- using a hardcoded height to account for the before content "ADVERTISEMENT"
Screen Shot 2022-08-02 at 17 21 51

#### Admin 
- noted available color differences in discrepancies
- stacked the text rather than odd margin right behavior inline
Screen Shot 2022-08-02 at 17 13 14

#### Reader view 
- Removed hardcoded height 
- Using paragraph component to add the label 
- Match color and styles based on body-small sizes 
- Centered because the width would need to be added to the paragraph container. The width of the ad could be dynamic 
- Updated discrepancies

Screen Shot 2022-08-02 at 17 22 02

## Dependencies or Side Effects

- feature pack pr https://github.com/WPMedia/arc-themes-feature-pack/pull/317
- manifest pr https://github.com/WPMedia/arc-themes-manifests/pull/119
- See discrepancies in the doc https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138945094/Blocks+Design+Discrepancies

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

# NOTES 

- [x] Start a storybook mock with the alias set 
- [x] Show something for the storybook
- [x] Determine if we need to convert admin-only styled components to enhanced styles -> yes, I think this will be easier to convert
- [x] Remove margins around the block on admin that seems hard-coded via classnames
- [x] Convert ArcAdminAd in admin view for showing the dimensions, slot name, etc, to v2 including sb
- [x] Show display label ad using v2 not styled components
- [x] ~~Use dangerously set html for display label ad?~~  -> nope. content css property doesn't support setting html. so this wouldn't work before (no pun intended) 

```js
		// todo: dangerously set for advertisement label? this is just setting the prop
		it("renders custom advertisement label", () => {
			const advertisementLabel =
				"Advertisement / <a href='http://example.com' target='_blank'>Advertisement</a>";

```

- [x] Add/update tests for display label ad
- [x] Set breakpoints in this block rather than in the site properties -> @nschubach updated this to not rely upon the hidden blocks.json breakpoints to do this
- [x] Remove magic number of 17 for the line height for the text. Set the min height on the ad element rather than the parent element that includes the paragraph tag
- [x] Remove unnecessary isAdmin check within ad unit show. Update tests
- [ ] Show ads in storybook or at least their space. Possible? Maybe placeholder? 
- [x] Update docs to remove placeholder info. Update to be more useful
- [x] Update the package.json for v2 -> excluded main entry-file update 
- [x] Validate that the entryfile still needs to be index.js -> in index.js file, there's a note to not remove the index.js file until ads block is not being imported elsewhere

## Questions 

- How do we want to handle the breakpoints in the site properties? Leave them because they are js based? -> Nick has this set :) 
- do we need to continue to support amp logic here? -> yes
- does advertisement label allow whatever html? not just text? -> does not allow whatever html. only text

## admin view 

have a styled component working in the commit https://5f91e4721ccaba0022a10782-qucvrsbzrw.chromatic.com/?path=/story/blocks-ads-block--admin-view
Screen Shot 2022-08-01 at 15 25 36
https://github.com/WPMedia/arc-themes-blocks/pull/1455/commits/c0829e95d2f4b49487540cb0799a4701dde9d521

-> will mimic this

### CODE REVIEW

File: blocks/ads-block/_index.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@use "@wpmedia/arc-themes-components/scss";
+
+.b-ads-block {
+	&--admin {
```
Comment (JackHowa) [inline blocks/ads-block/_index.scss:4]: wasn't sure if this needed a sub-element. it's really a modifier of the sub-element that will only exist if admin

File: blocks/ads-block/features/ads/_children/AdUnit/index.jsx [hunk 2]
```diff
@@ -27,7 +28,12 @@ const AdUnit = (props) => {
 		if (!isAdmin) registerAd();
 	}, [registerAd, isAdmin]);
 
-	return !isAdmin ? <div id={id} className={`arcad ad-${adClass}`} /> : null;
+	return !isAdmin ? (
```
Comment (JackHowa) [inline blocks/ads-block/features/ads/_children/AdUnit/index.jsx]: todo: remove isAdmin from this logic. this component will only be shown with admin

**Comment by nschubach** [comment]
Added PR for a breakpoint proposal: https://github.com/WPMedia/arc-themes-blocks/pull/1458

File: blocks/ads-block/features/ads/ad-helper.js [hunk 3]
```diff
@@ -8,6 +7,7 @@ export const getSizemapBreakpoints = (props) => {
 	return !breakpoints
 		? []
 		: [
+				// setup these multiple breakpoints in the theme styles breakpoints
```
Comment (nschubach) [inline blocks/ads-block/features/ads/ad-helper.js]: See: https://github.com/WPMedia/arc-themes-blocks/pull/1458

Comment (nschubach) [reply blocks/ads-block/features/ads/ad-helper.js:10]: I interpreted this need wrong.  This is not for styling?  Ignore this PR...

Comment (nschubach) [reply blocks/ads-block/features/ads/ad-helper.js:10]: this seem to be for populating `arcads` `sizemap` values.  Need to dig into `arcads` to find out what these are for.

Comment (nschubach) [reply blocks/ads-block/features/ads/ad-helper.js:10]: I think I have a solve for this, can talk about it in dev huddle

Comment (nschubach) [reply blocks/ads-block/features/ads/ad-helper.js]: ref: https://github.com/washingtonpost/arcads#size-mapping

File: blocks/ads-block/features/ads/_children/AdUnit/index.jsx (hunk 2 above)
Comment (JackHowa) [reply blocks/ads-block/features/ads/_children/AdUnit/index.jsx]: done

File: blocks/ads-block/features/ads/ad-helper.js (hunk 3 above)
Comment (JackHowa) [reply blocks/ads-block/features/ads/ad-helper.js]: thanks for your pr

**Review by matthewroach — APPROVED** [review]

File: blocks/ads-block/features/ads/ad-helper.js (hunk 3 above)
Comment (JackHowa) [reply blocks/ads-block/features/ads/ad-helper.js]: is this ok to resolve @nschubach ?

**Pull request merged**